controller/api/public/http_server.go

```diff
@@ -279,6 +306,21 @@ func (s tapServer) Send(msg *pb.TapEvent) error {
 	return nil
 }
 
+type destinationServer struct {
+	tapServer
```

Comment:
It feels strange to me that `destinationServer` embeds `tapServer`, since the destination and tap endpoints are unrelated. It might make sense to instead provide a `StreamServer` struct (a corollary to the `StreamClient` struct that you're creating in this branch), and then embed that struct in both `destinationServer` and `tapServer`.